src/main/scala/it/pagopa/interop/partymanagement/api/impl/PartyApiServiceImpl.scala

```diff
@@ -121,6 +121,88 @@ class PartyApiServiceImpl(
 
   }
 
+  /** Code: 200, Message: successful operation, DataType: Institution
+   * Code: 405, Message: Institution not found, DataType: Problem
+   * Code: 400, Message: Invalid ID supplied, DataType: Problem
+   */
+  override def updateInstitutionById(id: String, institution: Institution)(implicit
+    toEntityMarshallerInstitution: ToEntityMarshaller[Institution],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem],
+    contexts: Seq[(String, String)]
+  ): Route = {
+    logger.info(s"Updating institution $id")
+
+    val commander: EntityRef[Command] = getCommander(id)
+
+    val updatedInstitution: Future[StatusReply[Party]] = for {
+      uuid       <- id.toFutureUUID
+      found      <- commander.ask(ref => GetParty(uuid, ref))
+      party      <- found.toFuture(UpdateInstitutionNotFound(id)).map(InstitutionParty.extractInstitutionParty)
+      updatedOrg <- party
+        .map(p => {
+          if (p.externalId != institution.institutionId) {
+            Left(
+              Failure(
+                UpdateInstitutionBadRequest(
+                  id,
+                  s"Cannot update externalId ${p.externalId} -> ${institution.institutionId}"
+                )
+              )
+            )
+          } else {
+            Right(
+              InstitutionParty(
```

Comment:
A big chunk of this case class construction can be done with the `.copy()` method that overrides just the field you need to.